src/Core/Models/RegisterSystemRequest.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Altinn.Platform.Authentication.Core.Models;
+using System.ComponentModel.DataAnnotations;
+
+#nullable enable
+namespace Altinn.Platform.Authentication.Core.SystemRegister.Models
+{
+    /// <summary>
+    /// Model for the System Register
+    /// A Registered System is a product supplied by a System Vendor,
+    /// it may need Rights to use or acccess Resources at a Service Provider.
+    /// </summary>
+    public class RegisterSystemRequest
+    {
+        /// <summary>
+        /// The unique Id for this product, in human-readable string format.
+        /// The id is in the format of system_vendor_name_plus_name_chosen_by_them.
+        /// 
+        /// An Optimistic Concurrency pattern to create new System Ids is used,
+        /// where the Id of the product is prefixed with the SystemVendor to help with uniqueness.
+        /// 
+        /// When the SystemVendor tries to register a new system (product), 
+        /// they should be aware of their own previous system names 
+        /// when giving the new system it's id.
+        /// </summary>
+        [Required]
+        public string CustomSystemId { get; set; } = string.Empty;
```

Comment:
We should use one naming for this. Now we use productId somewhere and customSystemID other places

Reply:
Yes, Also on the route. Can we agree on a naming here. Can we call the system as "product"? Because i see we have that in the url and then we are using the term system and product interchangeably everywhere